app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt

```diff
@@ -15,48 +15,59 @@ import com.nilhcem.blenamebadge.R
 
 import java.math.BigInteger
 import java.util.ArrayList
+import android.animation.ValueAnimator
+import android.view.animation.LinearInterpolator
+import com.nilhcem.blenamebadge.device.model.Mode
+import com.nilhcem.blenamebadge.device.model.Speed
 
 class PreviewBadge : View {
-    private var ledDisabled: Drawable? = null
-    private var ledEnabled: Drawable? = null
+    private var ledDisabled: Drawable
+    private var ledEnabled: Drawable
 
     private lateinit var bgBounds: RectF
     private var cells = ArrayList<Cell>()
 
     private var badgeHeight = 11
     private var badgeWidth = 44
 
-    private var checkList: ArrayList<CheckList>? = null
+    private var oneByte = 8
+
+    private var ifFlash: Boolean = false
+    private var ifMarquee: Boolean = false
+    private var badgeSpeed: Int = 1
+    private var badgeMode: Mode = Mode.LEFT
+
+    private var animationIndex: Int = 0
+
+    private lateinit var checkList: ArrayList<CheckList>
```

Comment:
Same question, why lateinit at all? Why can't it be initialized here?

Reply:
Alright Sir